### PR TITLE
Pal 146 data service testing verification

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java [hunk 1]
```diff
@@ -81,7 +80,7 @@
     private ResponseEntity<Object> noPolicyExceptionHandler(final NoPolicyException ex, final WebRequest request) {
         ErrorDetails details = new ErrorDetails(ZonedDateTime.now(), MESSAGE, ex.getMessage(), ex.getStackTrace());
 
-        LOGGER_DEBUG.error("Error from data service, details: {}, and request {}", details, request);
+        LOGGER.error("Error from data service, details: {}", details);
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java]: It is useful to have the request available that caused the exception
```suggestion
       LOGGER.error("Error from data service, details: {}, and request {}", details, request);
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java [hunk 2]
```diff
@@ -94,7 +93,7 @@
     private ResponseEntity<Object> nullPointerExceptionHandler(final NullPointerException ex, final WebRequest request) {
         ErrorDetails details = new ErrorDetails(ZonedDateTime.now(), MESSAGE, ex.getMessage(), ex.getStackTrace());
 
-        LOGGER_DEBUG.error("Error from data service, details: {}, and request {}", details, request);
+        LOGGER.error("Error from data service, details: {}", details);
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java]:
```suggestion
        LOGGER.error("Error from data service, details: {}, and request {}", details, request);
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java [hunk 3]
```diff
@@ -107,7 +106,7 @@
     private ResponseEntity<Object> fileNotFoundExceptionHandler(final FileNotFoundException ex, final WebRequest request) {
         ErrorDetails details = new ErrorDetails(ZonedDateTime.now(), MESSAGE, ex.getMessage(), ex.getStackTrace());
 
-        LOGGER_DEBUG.error("Error from data service, details: {}, and request {}", details, request);
+        LOGGER.error("Error from data service, details: {}", details);
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java]:
```suggestion
        LOGGER.error("Error from data service, details: {}, and request {}", details, request);
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java [hunk 4]
```diff
@@ -120,7 +119,7 @@
     private ResponseEntity<Object> ioExceptionHandler(final IOException ex, final WebRequest request) {
         ErrorDetails details = new ErrorDetails(ZonedDateTime.now(), MESSAGE, ex.getMessage(), ex.getStackTrace());
 
-        LOGGER_DEBUG.error("Error from data service, details: {}, and request {}", details, request);
+        LOGGER.error("Error from data service, details: {}", details);
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java]:
```suggestion
        LOGGER.error("Error from data service, details: {}, and request {}", details, request);
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java [hunk 5]
```diff
@@ -133,7 +132,7 @@
     private ResponseEntity<Object> runtimeExceptionHandler(final RuntimeException ex, final WebRequest request) {
         ErrorDetails details = new ErrorDetails(ZonedDateTime.now(), MESSAGE, ex.getMessage(), ex.getStackTrace());
 
-        LOGGER_DEBUG.error("Error from data service, details: {}, and request {}", details, request);
+        LOGGER.error("Error from data service, details: {}", details);
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java]:
```suggestion
        LOGGER.error("Error from data service, details: {}, and request {}", details, request);
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java [hunk 6]
```diff
@@ -146,7 +145,7 @@
     private ResponseEntity<Object> globalExceptionHandler(final Exception ex, final WebRequest request) {
         ErrorDetails details = new ErrorDetails(ZonedDateTime.now(), MESSAGE, ex.getMessage(), ex.getStackTrace());
 
-        LOGGER_DEBUG.error("Error from data service, details: {}, and request {}", details, request);
+        LOGGER.error("Error from data service, details: {}", details);
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/ExceptionHandler.java]:
```suggestion
        LOGGER.error("Error from data service, details: {}, and request {}", details, request);
```

File: data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java [hunk 7]
```diff
@@ -134,10 +143,11 @@ private void auditRequestReceivedException(final ReadRequest request, final Thro
...
             LOGGER.info("Calling dataReader with: {}", readerRequest);
 
-            final DataReaderResponse readerResult = getDataReader().read(readerRequest,
-                    this.getClass(),
-                    auditRequestReceiver);
+            AtomicLong recordsProcessed = new AtomicLong(0);
+            AtomicLong recordsReturned = new AtomicLong(0);
+            final DataReaderResponse readerResult = getDataReader().read(readerRequest, recordsProcessed, recordsReturned);
             LOGGER.info("Reader returned: {}", readerResult);
+            LOGGER.info("Processed {} and returned {} records", recordsProcessed.get(), recordsReturned.get());
```
Comment (nw1984) [inline data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java]: Combine the two info messages:
```suggestion
            LOGGER.info("Processed {} and returned {} records, reader returned: {}", recordsProcessed.get(), recordsReturned.get(), readerResult);
```

**Review by nw1984 — DISMISSED** [review]

**Review by ac74475 — DISMISSED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**